### PR TITLE
Use smart link component to handle php pages links

### LINKED ISSUE #1600: Create custom link component to handle navigation to PHP pages
From https://github.com/iFixit/react-commerce/issues/1594#issuecomment-1513423042

> By looking at our production Strapi, I came up with an idea. I saw that we're using relative paths for internal Next.js links, and absolute paths for php pages. I'm proposing to create a custom link component that uses a Next.js link for relative paths, and a normal link for absolute paths. This way we can have catch all routes, and still link to php pages without issues.

### PR DESCRIPTION
**Pull request by dhmacs**
closes #1600 

## QA

1. Open a [product list preview](https://react-commerce-git-add-smart-link-component-ifixit.vercel.app/Parts/Android_Phone) and verify these links are working:
    1. Main navigation menu
    2. Product list breadcrumb
    3. Secondary nav links (guides and answers)
    4. Page edit menu links (admin only)
    5. Product list categories links 
6. Open a [product page preview](https://react-commerce-git-add-smart-link-component-ifixit.vercel.app/products/iphone-12-12-pro-replacement-battery) and verify these links are working:
    1. Breadcrumbs
    2. Device compatibility links

### CODE REVIEW

**Comment by dhmacs** [comment]
Some jobs are failing, and it looks like something is not working in the cominor api

Screenshot 2023-04-20 at 09 19 00

cc @ardelato @sterlinghirsh @masonmcelvain

**Comment by deltuh-vee** [comment]
Do we expect the Device Compatibility links to go to cominor?

**Comment by sterlinghirsh** [comment]
> Do we expect the Device Compatibility links to go to cominor?

I would have expected so. I just checked them and they are pointing to urls like https://react-commerce-git-add-smart-link-component-ifixit.vercel.app/Device/iPhone_12_Pro, however the same issue is present on main (i.e. from https://react-commerce.vercel.app/products/iphone-12-12-pro-replacement-battery) so unless you're referring to something else, I think it's ok. I think it should work fine when viewing through cominor or ifixit.

**Comment by dhmacs** [comment]
@deltuh-vee @sterlinghirsh They are relative links, but they're rendered as a plan html `a`, so they would work fine in production

Screenshot 2023-04-21 at 08 10 29

**Comment by deltuh-vee** [comment]
dev_block ✊ 
image
Answers link is going to `/Device/<device>` instead of `/Answers/Device/<device>`

**Comment by dhmacs** [comment]
@deltuh-vee fixed 👍 

un_dev_block 🟢

**Comment by deltuh-vee** [comment]
QA 🎬 
Links are working.

File: frontend/components/ui/SmartLink.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { Box, BoxProps as ChakraBoxProps, forwardRef } from '@chakra-ui/react';
+import NextLink, { LinkProps } from 'next/link';
+
+type SmartLinkProps = Pick<LinkProps, 'legacyBehavior'> &
+   ChakraBoxProps & {
+      href: string;
+      behaviour?: LinkBehavior;
+   };
+
+type LinkBehavior = 'auto' | 'reload' | 'fast';
```
Comment (masonmcelvain) [inline frontend/components/ui/SmartLink.tsx:10]: What does 'fast' do?

**Review by masonmcelvain — APPROVED** [review]
CR ✌🏻 Nice work!

File: frontend/components/ui/SmartLink.tsx (hunk 1 above)
Comment (dhmacs) [reply frontend/components/ui/SmartLink.tsx:10]: It's supposed to represent the normal Next.js behaviour, it should behave like just using a Next.js link, although practically speaking this component should probably be used only with link behaviour `auto`.

**Pull request merged**